test/e2e/catalog_e2e_test.go

```diff
@@ -761,3 +767,98 @@ func replicateCatalogPod(t *testing.T, c operatorclient.ClientInterface, crc ver
 
 	return copied
 }
+
+func TestGrpcConnectionStateIsNeverTransient(t *testing.T) {
+	// Create gRPC CatalogSource using an external registry image (community-operators)
+	// Wait for a registry pod to be created
+	// Monitor status and report failure if transitions do not progress cleanly (connecting -> connected)
+	// Delete the registry pod (part of cleaning up)
+	defer cleaner.NotifyTestComplete(t, true)
+	sourceName := genName("catalog-")
+
+	// Create gRPC CatalogSource using an external registry image (community-operators)
+	source := &v1alpha1.CatalogSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       v1alpha1.CatalogSourceKind,
+			APIVersion: v1alpha1.CatalogSourceCRDAPIVersion,
+		},
```

Comment:
Are there issues if the `TypeMeta` isn't added up front?

Reply:
No, I just copied it from another test.